### PR TITLE
Shortcut import process to speed up duplicate imports

### PR DESCRIPTION
**Pull request by peppy**
We don't have a good benchmark process in place so I just did some quick tests using import tests (debug):

# Importing same beatmap 1000 times

Smgi: 9.5s (#13666)
New: 15.6s (this PR)
Old: 305s (master)

Note that these numbers include startup overhead of gamehost/game; actual experience performance will be higher.

- [x] Depends on #13672.

### CODE REVIEW

File: osu.Game/Database/ArchiveModelManager.cs [hunk 1]
```diff
@@ -309,6 +309,12 @@ internal static void LogForModel(TModel model, string message, Exception e = nul
                 Logger.Log($"{prefix} {message}", LoggingTarget.Database);
         }
 
+        /// <summary>
+        /// Whether the implementation overrides <see cref="ComputeHash"/> with a custom implementation.
+        /// Custom has implementations must bypass the early exit in the import flow (see <see cref="computeHashFast"/> usage).
```
Comment (frenzibyte) [inline osu.Game/Database/ArchiveModelManager.cs:314]:
```suggestion
        /// Custom hash implementations must bypass the early exit in the import flow (see <see cref="computeHashFast"/> usage).
```

**Comment by bdach** [comment]
Looks like the failing import tests need to be pretty much removed with this behaviour change.

**Comment by peppy** [comment]
All tests were tested at my end to pass. Will double-check.

**Comment by peppy** [comment]
Missed some stashed changes. Two of the tests are intentionally regressed; the other failures were a missing `Undelete()` call.

**Review by smoogipoo — APPROVED** [review]

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**